hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngineImpl.java

```diff
@@ -88,8 +88,7 @@ public Result execute(Query query, Target target) {
         } else if (target.isTargetLocalNode()) {
             return runQueryOnLocalPartitions(adjustedQuery);
         } else if (target.isTargetPartitionOwner()) {
-            // we do not adjust the query here - it's a single partition operation only
```

Comment:
I am not quite sure why there was an exception. possible it was just an optimization. 

however when adjustment is not applied then e.g. `values(pagingPreciate)` internally does not fetch keys and as a consequence paging anchor does not work.